lib/bintray_client.rb

```diff
@@ -0,0 +1,40 @@
+require_relative './http_client.rb'
+
+class BintrayClient
+
+  def initialize(user:, key:, base_url: 'https://bintray.com/api/v1/packages')
+    @base_url = base_url
+    @user = user
+    @key = key
+    @http_client = HTTPClient.new
```

Comment:
Shouldn't you pass the user/key to the `HTTPClient.new`, since it's using it in its impl (see below)?
Or if you keep it at `BintrayClient` level (which might make more sense), you'll need to pass them as parameters to `delete_json`